posydon/utils/ignorereason.py

```diff
@@ -0,0 +1,48 @@
+"""Class for handling ignore reasons when creating new grids."""
+
+
+IGNORE_REASONS_PRIORITY = [
+    'corrupted_binary_history',
+    'corrupted_history1',
+    'corrupted_history2',
+    'ignored_no_BH',
+    'ignore_no_H1',
+    'ignored_scrubbed',
+    'ignore_no_FP',
+    'ignored_no_RLO']
```

Comment:
I suggest to change the priority order. E.g. having no history should have a higher priority to having a corrupted history.
Additionally, I'd like to think about aligning then reasons more, e.g. using the short version BH vs. the long version binary_history or whether we need the key word `ignore` in this class anymore.
Finally, I'd suggest to add already some more pre defined reasons for future use, e.g. `ignore_no_H2` and/or being more specific by splitting `ignored_scrubbed` into `ignored_scrubbed_BH`, `ignored_scrubbed_H1`, `ignored_scrubbed_H2`.
So, I'd suggest:
```
IGNORE_REASONS_PRIORITY = [
    # missing data files
    'no_history1',              # always needed
    'no_binary_history',        # for all binaries needed, while not for single stars
    'no_history2',              # needed, if there are two stars in the binary
    # corrupted data files
    'corrupted_history1',       # always needed
    'corrupted_binary_history', # for all binaries needed, while not for single stars
    'corrupted_history2',       # needed, if there are two stars in the binary
    # scrubbed data
    'scrubbed_history1',        # always needed
    'scrubbed_binary_history',  # for all binaries needed, while not for single stars
    'scrubbed_history2',        # needed, if there are two stars in the binary
    # issues with final profile data
    'no_final_profile',         # missing file
    'corrupted_final_profile',  # corrupted file
    'scrubbed_final_profile',   # scrubbed data
    # not needed data
    'no_RLO']
```

Reply:
@mkruckow I will change the order of the errors as you mentioned. I am reluctant to put flags that do not exist yet in `psygrid` because there might be reasons for that. For example, the `scrubbed_history1` is not necessary because for a scrubbing history issue, we don't really need to have this resolution - it's a complicated problem that needs manually checking all history files, and it may be the case that don't know where the issue is (e.g., if you have a binary history and a history file with no overlap in the steps... you don't know which one if off).

I will change some names for consistency though - thanks for inspiring this.

Reply:
You introduced some new inconsistencies: e.g. 'ignored_no_H1' vs. 'corrupted_history1'.

Reply:
If you mean about `H1` vs `history1` I fixed that now, thanks.

Reply:
Mainly this, yes. But I'd leave out the "ignored" as well or put it everywhere, because all reasons are reason to have something ignored.